### PR TITLE
Add ANSI colors to luma.core.virtual.terminal

### LINKED ISSUE #40: Add ANSI colors to luma.core.virtual.terminal
`puts()` processes a string of characters one at time, but this could be changed to be made aware of ANSI escape sequences, particularly the color codes below.

On encountering a new (valid) color code, the `self.color` and/or `self.bgcolor` could be set accordingly. Also need to consider what the reset code (`ESC[0m`) would reset the colors to.

screen shot 2017-03-06 at 21 55 35

(screen grab taken from http://ascii-table.com/ansi-escape-sequences.php)

### PR DESCRIPTION
**Pull request by rm-hull**
Fixes #40

### CODE REVIEW

File: luma/core/ansi_color.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Richard Hull and contributors
+# See LICENSE.rst for details.
+
```
Comment (thijstriemstra) [inline luma/core/ansi_color.py]: can you add a docstring?

File: luma/core/ansi_color.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    42:  ["background_color", "green"],
+    43:  ["background_color", "yellow"],
+    44:  ["background_color", "blue"],
+    45:  ["background_color", "magenta"],
+    46:  ["background_color", "cyan"],
+    47:  ["background_color", "white"],
+}
+
+
+def parse_str(text):
```
Comment (thijstriemstra) [inline luma/core/ansi_color.py]: parse_ansi_str?

File: luma/core/ansi_color.py (hunk 1 above)
Comment (thijstriemstra) [inline luma/core/ansi_color.py]: and a version added directive in that docstring.

File: luma/core/ansi_color.py (hunk 2 above)
Comment (rm-hull) [inline luma/core/ansi_color.py]: In context of `ansi_color.parse_ansi_str`, it seems a bit of a mouthful. 
In isolation `parse_ansi_str` is probably ok, but generally: what benefit/drawback does adding 'ansi' into the method name convey?

**Pull request merged**